consumers/writers/api/logging.go

```diff
@@ -13,30 +13,30 @@ import (
 	log "github.com/mainflux/mainflux/logger"
 )
 
-var _ consumers.Consumer = (*loggingMiddleware)(nil)
+var _ consumers.BlockingConsumer = (*loggingMiddleware)(nil)
 
 type loggingMiddleware struct {
-	logger   log.Logger
-	consumer consumers.Consumer
+	logger       log.Logger
+	syncConsumer consumers.BlockingConsumer
```

Comment:
Use simply `consumer` instead of `syncConsumer`. Also `func(lm *loggingMiddleware)` instead of `func(slm *loggingMiddleware)` as a receiver.